.github/workflows/conda.yml

```diff
@@ -24,9 +24,25 @@ defaults:
     shell: bash -l {0}
 
 jobs:
+  check_tag:
+    runs-on: ubuntu-latest
+    outputs:
+      is_tagged_commit: ${{ steps.check_tag.outputs.is_tagged_commit }}
+    steps:
+      - uses: actions/checkout@v4.1.5
+      - name: Check if the latest commit is tagged
+        id: check_tag
+        run: |
+          if git describe --tags --exact-match HEAD > /dev/null 2>&1; then
+            echo "is_tagged_commit=true" >> $GITHUB_OUTPUT
+          else
+            echo "is_tagged_commit=false" >> $GITHUB_OUTPUT
+          fi
   conda:
     name: Build (and upload)
     runs-on: ubuntu-latest
+    needs: check_tag
+    if: startsWith(github.ref, 'refs/tags/') || needs.check_tag.outputs.is_tagged_commit == 'false'
```

Comment:
So this needs to be a tag or not? Is there a particular case we are trying to exclude here?

Reply:
_TLDR - this filters out branch push-triggered runs if we see the push was associated with a tag, since we already know this workflow was triggered by the corresponding tag push_

Since https://github.com/dask/dask/pull/11014 there's been a race condition on release (which coincides with the push of a tagged commit) where this workflow is triggered twice for a `branch` and `tag` push and attempts to upload the same package twice. Consider these workflows for the 2024.8.1 release, one passing and one failing at the upload stage:

- https://github.com/dask/dask/actions/runs/10426906155/job/28880711510
- https://github.com/dask/dask/actions/runs/10426906162/job/28880711395

There are probably a few different ways to handle something like this, but I opted to check if the current `HEAD` is an exact match with a tag as a filtering condition to stop `branch`-triggered runs since at that point we know there's a corresponding `tag` triggered run, which we can now treat as the "definitive" run of this workflow on a release.